PR #1971 needs test update for error-line tolerance change

## Problem

PR #1971 (for issue #1951) introduces a tolerance-based error-line comparison but doesn't update the test that verifies new error detection.

## Test Failure

```
FAILED tests/shepherd/test_phases.py::TestBuilderFallbackComparison::test_fallback_new_errors_detected
```

The test expects 1 new error line to be detected as a failure, but the new `_ERROR_LINE_TOLERANCE = 5` treats this as pre-existing (non-deterministic noise).

## Current Test

```python
baseline_result = subprocess.CompletedProcess(
    args=[], returncode=1, stdout="Error: old bug\n", stderr=""
)
worktree_result = subprocess.CompletedProcess(
    args=[],
    returncode=1,
    stdout="Error: old bug\nError: new regression\n",  # 1 new line
    stderr="",
)
```

Difference = 1 line, which is ≤ tolerance of 5, so treated as pre-existing.

## Fix Needed

Update the test to use a difference greater than the tolerance:

```python
# Use 7 error lines (6 more than baseline) to exceed tolerance of 5
worktree_result = subprocess.CompletedProcess(
    args=[],
    returncode=1,
    stdout="Error: old bug\n" + "".join(f"Error: new regression {i}\n" for i in range(6)),
    stderr="",
)
```

## Context

- Original issue: #1951
- PR: #1971
- Review comment: https://github.com/rjwalters/loom/pull/1971#issuecomment-3831213655

This is blocking PR #1971 from being approved.

--- a/loom-tools/tests/shepherd/test_phases.py
+++ b/loom-tools/tests/shepherd/test_phases.py
@@ -2248,7 +2248,11 @@ class TestBuilderFallbackComparison:
     def test_fallback_new_errors_detected(
         self, mock_context: MagicMock
     ) -> None:
-        """Fallback: new error lines in worktree -> FAILED."""
+        """Fallback: new error lines in worktree -> FAILED.
+
+        Uses 7 error lines in worktree vs 1 in baseline (diff=6) to exceed
+        the _ERROR_LINE_TOLERANCE of 5, ensuring new errors are detected.
+        """
         builder = BuilderPhase()
         worktree_mock = MagicMock()
         worktree_mock.is_dir.return_value = True
@@ -2257,10 +2261,12 @@ class TestBuilderFallbackComparison:
         baseline_result = subprocess.CompletedProcess(
             args=[], returncode=1, stdout="Error: old bug\n", stderr=""
         )
+        # 7 error lines total (1 original + 6 new) exceeds tolerance of 5
+        new_errors = "".join(f"Error: new regression {i}\n" for i in range(6))
         worktree_result = subprocess.CompletedProcess(
             args=[],
             returncode=1,
-            stdout="Error: old bug\nError: new regression\n",
+            stdout=f"Error: old bug\n{new_errors}",
             stderr="",
         )
         with (

